File: node/src/api/arch/corp_coop.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/06/08
 */

import { ApiClient } from '../../base/api_client';
import { CORP_COOP_BASE_PATH } from '../../constants/http_constant'

export class CorpCoopApiClient extends ApiClient {
    validVersions = ['v5'];
    defaultVersion = 'v5';

    /**
     * check version
     */
    async doRequest(options) {
        if (!this.validVersions.includes(this.version)) {
            this.version = this.defaultVersion;
        }
        return super.doRequest(options);
    }

    /**
     * 列出我连接的企业
     */
    async corpCoopDepartList(options) {
        return await this.doRequest({
            method: 'POST',
            path: CORP_COOP_BASE_PATH + 'department/list',
            payload: {
                dept_no: options.deptNo
            }
        });
    }

    /**
     * 列出我连接的企业对接人
     */
    async corpCoopMemberList(options) {
        return await this.doRequest({
            method: 'POST',
            path: CORP_COOP_BASE_PATH + 'user/list',
            payload: {
                dept_no: options.deptNo
            }
        });
    }

    /**
     * 列出我连接的企业对接人详细信息
     */
    async corpCoopUserInfo(username) {
        return await this.doRequest({
            method: 'POST',
            path: CORP_COOP_BASE_PATH + 'user/get',
            payload: {
                username
            }
        });
    }
}

File: python/src/api/arch/corp_coop.py
from src.model.http.api_client import ApiClient
from ...constants.http_constant import HttpConstant
from ...model.http.http_request_param import HttpRequestParam

# 合法的版本
valid_versions = ('v5')
# 默认版本
default_version = 'v5'


class CorpCoopApiClient(ApiClient):

    def __init__(self, api_key, host):
        ApiClient.__init__(self, api_key, host, valid_versions, default_version)

    '''
    列出我连接的企业
    Arguments:
        dept_no: integer 部门编号
        version: 版本
    '''

    def corpCoopDepartList(self, dept_no, version=default_version):
        path = HttpConstant.CORP_COOP_PATH.format(suffix='department/list',
                                                  version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, {"dept_no": dept_no})
        return self.send_post(request_param)

    '''
    列出我连接的企业对接人
    Arguments:
        dept_no: integer 部门编号
        version: 版本
    '''

    def corpCoopMemberList(self, dept_no, version=default_version):
        path = HttpConstant.CORP_COOP_PATH.format(suffix='user/list',
                                                  version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, {"dept_no": dept_no})
        return self.send_post(request_param)

    '''
    列出我连接的企业对接人详细信息
    Arguments:
        user_name: str 用户名
        version: 版本
    '''

    def corpCoopUserInfo(self, user_name, version=default_version):
        path = HttpConstant.CORP_COOP_PATH.format(suffix='user/get',
                                                  version=self.getValidVersion(version))
        request_param = HttpRequestParam(path, {"username": user_name})
        return self.send_post(request_param)
